File: 2706-buy-two-chocolates/2706-buy-two-chocolates.js
var buyChoco = function(prices, money) {
    let mini = 101;
    let mini2 = Infinity;
    let index = 0;

    for (let i = 0; i < prices.length; i++) {
        if (mini > prices[i]) {
            index = i;
            mini = prices[i];
        }
    }
    
    for (let i = 0; i < prices.length; i++) {
        if (i !== index) {
            mini2 = Math.min(mini2, mini + prices[i]);
        }
    }
    
    return (mini2 <= money ? money - mini2 : money);
};

File: 2706-buy-two-chocolates/2706-buy-two-chocolates.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        first_min = second_min = maxsize

        for p in prices:
            if p < first_min:
                second_min, first_min = first_min, p
            elif p < second_min:
                second_min = p

        min_price = first_min + second_min
        return money - min_price if min_price <= money else money